client/my-sites/stats/overview.jsx

```diff
@@ -82,9 +84,14 @@ class StatsOverview extends Component {
 			);
 		} );
 
+		// Generate the current date in yyyy-mm-dd format, in UTC.
+		const date = new Date().toISOString().split( 'T' )[ 0 ];
```

Comment:
Are we considering the current timezone of the current site when doing this? Could there be any possible issues given the site's potential timezone offset from UTC?

Reply:
The endpoint does calculate each sites' timezone offset and use that when building the queries used to calculate data. That said, I'm still somewhat unclear if something is being missed here. It could be the case.
I think a potential failure could be if UTC and local dates are different, the user sees the text for the UTC date on their screen (but the stats are calculated the same as before). It needs a closer look after we're happy with the data layer.

Reply:
There were issues with the timezone, but I have resolved them now. See lower in the main thread for details.